tasks/init.rb

```diff
@@ -80,13 +81,54 @@ def request(method, uri, headers, body, opts)
     )
   end
 
+  # Parses the response body.
+  def parse_response_body(response, json)
+    body = encode_body(response.read_body, response.type_params['charset'])
+
+    if json
+      begin
+        body = JSON.parse(body)
+      rescue JSON::ParserError => e
+        raise TaskHelper::Error.new(
+          "Unable to parse response body as JSON: #{e.message}",
+          'http_request/json-parse-error'
+        )
+      end
+    end
+
+    body
+  end
+
   # Forces the response body to the specified encoding and
   # then encodes as UTF-8.
   def encode_body(body, charset)
     body = body.force_encoding(charset) if charset
     body.encode('UTF-8')
   end
 
+  # Formats the body. If the request is a JSON request, this will
+  # convert the body to JSON.
+  def format_body(body, json)
+    if json && !body.is_a?(String) && !body.nil?
+      body.to_json
```

Comment:
I think this is conflating *sending* JSON to the endpoint with *receiving* JSON from an endpoint. It should be up to the task user to ensure they're sending valid JSON if the endpoint needs it, this PR should just be to ensure the response is JSON if the task author needs to parse the response.

Reply:
Here's the related issue that calls for the `json_endpoint` parameter to result in the task setting headers automatically and serializing + parsing the body as JSON: https://github.com/puppetlabs/bolt/issues/2103

I think adding in the extra step of requiring a task user to not only serialize the body as JSON but also manually set the headers is a little unfriendly and kind of defeats the purpose of creating a parameter that will handle JSON API requests for you... at that point why not just create a `parse_json` function in `puppetlab/stdlib` instead of adding a `json_endpoint` parameter?

Reply:
Hm ok - I'm (maybe irrationally) nervous that someone will try to use a poorly designed api that doesn't accept JSON but does return it, and it'd be a regression to have to add another parameter to do one and not the other. But if we're ok just not supporting badly designed APIs, that seems reasonable.